File: Leetcode/PY/Leetcode477-汉明距离总和.py
from typing import List
class Solution:
    def totalHammingDistance(self, nums: List[int]) -> int:
        n = len(nums)
        res = 0
        for i in range(30):
            c = 0
            for num in nums:
                if num >> i & 1 == 1:
                    c += 1
            res += c * (n - c)
        return res

File: Leetcode/Rust/Leetcode477-汉明距离总和.rs
impl Solution {
    pub fn total_hamming_distance(nums: Vec<i32>) -> i32 {
        let (mut res,n) = (0,nums.len() as i32);
        for i in 0..30{
            let mut c = 0;
            for num in nums.iter(){
                if num >> i & 1 == 0{
                    c += 1
                }
            }
            res += c * (n - c)
        }
        res
    }
}
